Packages/com.unity.inputsystem/InputSystem/Controls/IOptimizedControl.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace UnityEngine.InputSystem.Controls
+{
+    public enum OptimizedControlDataType
+    {
+        // If control state cannot be casted to any of types below due to various reasons
+        NotOptimized = 0,
+
+        // Bool value as 8 bit byte, byte aligned, where 0 means false and any other value means true
+        BoolAsByte,
+        
+        // Integer as 32 bit, byte aligned
+        Int32,
+        
+        // Float as 32 bit, byte aligned
+        Float,
+        
+        // Vector2 as 2 consecutive 32 bit floats, in order x,y, byte aligned
+        Vector2,
+
+        // Vector3 as 3 consecutive 32 bit floats, in order x,y,z, byte aligned
+        Vector3,
+
+        // Quaternion as 4 consecutive 32 bit floats, in order of x,y,z,w, byte aligned
+        Quaternion,
+
+        // PoseState
+        Pose
+    }
+    
+    internal interface IOptimizedControl
+    {
+        public OptimizedControlDataType CalculateOptimizedControlDataType();
+    }
```

Comment:
Could we get away with an isOptimized property on the base InputControl class instead of the interface?

Reply:
No, we need to cache the value somehow, because for example for `Vector3Control` to decide that it can be optimized it needs to know if x/y/z controls are optimized and are of float type

Reply:
N/A, interface removed